src/python/dxpy/workflow_builder.py

```diff
@@ -346,11 +396,29 @@ def _assert_executable_regions_match(workflow_enabled_regions, workflow_spec):
     in only one region - the region in which the applets are stored.
     """
     executables = [i.get("executable") for i in workflow_spec.get("stages")]
+    requested_regions = list(workflow_enabled_regions)
 
+    if 'billTo' in workflow_spec:
+        billable_regions = set()
+        try:
+            if workflow_spec['billTo'].startswith('user-'):
+                billable_regions = set(dxpy.api.user_describe(workflow_spec['billTo'])['permittedRegions'])
+            elif workflow_spec['billTo'].startswith('org-'):
+                billable_regions = set(dxpy.api.org_describe(workflow_spec['billTo'])['permittedRegions'])
+        except:
+            pass
+        
+        if billable_regions:
+            workflow_enabled_regions.intersection_update(billable_regions)
+    
     for exect in executables:
-
-        if exect.startswith("applet-") and len(workflow_enabled_regions) > 1:
-            raise WorkflowBuilderException("Building a global workflow with applets in more than one region is not yet supported.")
+        if exect.startswith("applet-"):
+            if len(workflow_enabled_regions) > 1:
```

Comment:
This same thing is getting checked multiple times, for each executable?

Reply:
I guess it will stop on the first applet if the exception will be thrown. Probably could be more nicely written as if len(workflow_enabled_regions) > 1 and filter(check it startsWith applet-) returns a non-empty result, then throw the exception. If only 1 region, the left side of the "and" will be false, and no need to proceed to the more computationally expensive right side of the "and."

Reply:
I changed it because it should be allowed to assign a single region where the applet is. But the old code is throwing an error.

Reply:
But if we request one region that does not contain the applet, checking the `len(workflow_enabled_regions)` could not detect the conflict.

Reply:
for example if the `workflow_enabled_regions` is `azure:us-west` but the `applet-xxxx` is in `aws:us-west`, the previous code will not detect it.